watertap/edb/db_api.py

```diff
@@ -276,34 +275,43 @@ def get_reactions(
             # {$not: {$elemMatch: { $nin: [<components>] } } } on that array
             stoich_field = Reaction.NAMES.stoich
             for item in collection.find():
+                stoich = {}
+                disallow = False
                 for phase in item[stoich_field].keys():
                     if allow_phases is not None and phase not in allow_phases:
-                        continue
-                    stoich = item[stoich_field][phase]
-                    if any_components:
-                        # look for non-empty intersection
-                        if set(stoich.keys()) & cnames:
-                            found.append(item)
+                        disallow = True
+                    for n in item[stoich_field][phase]:
+                        stoich[n] = item[stoich_field][phase][n]
+                #If the item involves a phase that is not allowed, then move on to next item
+                if (disallow):
```

Comment:
minor syntax point: parens are not needed here

Reply:
Noted. I can clean this up later. That way it doesn't automatically disregard your current review.